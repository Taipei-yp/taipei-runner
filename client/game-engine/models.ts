/** Координаты */
export type Coords = {
  x: number;
  y: number;
};

/** Размеры */
export type Sizes = {
  width: number;
  height: number;
};

/** Координаты с шириной */
export type CoordsAndWidth = Coords & { width: number };

/** Область пересечения */
export class CollisionBox {
  x: number;
  y: number;
  width: number;
  height: number;

  constructor(x: number, y: number, w: number, h: number) {
    this.x = x;
    this.y = y;
    this.width = w;
    this.height = h;
  }
}

/** Тип препятствия */
export type ObstacleType = {
  /** Название */
  type: string;
  /** Размеры в спрайте */
  sizes: Sizes;
  /** Начальная точка в спрайте */
  pos: Coords;
  /** Speed at which multiples are allowed */
  multipleSpeed: number;
  /** Minimum speed which the obstacle can make an appearance */
  minSpeed: number;
  /** minimum pixel space betweeen obstacles */
  minGap: number;
  collisionBoxes: CollisionBox[];
};

/** Тип линии горизонта */
export type HorizontLineType = {
  type: string;
  sizes: Sizes;
  spriteCoords: Coords;
  groundYMargin: number;
};

/** Набор фреймов и время отображения фрейма */
export type FrameSetType = {
  msPerFrame: number;
  frames: CoordsAndWidth[];
};
