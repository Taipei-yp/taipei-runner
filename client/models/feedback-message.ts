export type FeedbackMessage = {
  createdAt: number;
  topic: string;
  message: string;
};
