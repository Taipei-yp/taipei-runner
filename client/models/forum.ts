import { User } from "./user";

export type Message = {
  id: number;
  text: string;
  author: User;
  createdAt: string;
  reply: Message[] | undefined;
  parent_id: number;
  likes: number;
};

export type Topic = {
  id: number;
  name: string;
  createdAt: string;
  message: Message;
};
