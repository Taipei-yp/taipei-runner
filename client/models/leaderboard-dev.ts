type GameResult = {
  data: {
    userName: string;
    taipeidev: number;
  };
};

type GameResultForAPI = GameResult & {
  ratingFieldName: "taipeidev";
};

type GetLeaderboardDataRequest = {
  ratingFieldName: "taipeidev";
  cursor: number;
  limit: number;
};

export { GameResultForAPI, GetLeaderboardDataRequest };
