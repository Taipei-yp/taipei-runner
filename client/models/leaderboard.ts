type GameResult = {
  data: {
    userName: string;
    taipeiscore: number;
  };
};

type GameResultForAPI = GameResult & {
  ratingFieldName: "taipeiscore";
};

type GetLeaderboardDataRequest = {
  ratingFieldName: "taipeiscore";
  cursor: number;
  limit: number;
};

type GameResults = GameResult[];

export { GameResultForAPI, GameResults, GetLeaderboardDataRequest };
