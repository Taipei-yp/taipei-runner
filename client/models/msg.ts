export type Msg = {
  id: number;
  author: {
    firstName: string;
    lastName: string;
    avatar: string;
  };
  createdAt: string;
  text: string;
};
