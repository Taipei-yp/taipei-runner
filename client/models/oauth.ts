type OauthCode = { code: string };
type ServiceId = { service_id: string };

export { OauthCode, ServiceId };
