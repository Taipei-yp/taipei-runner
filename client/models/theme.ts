type Theme = {
  id: number;
  name: string;
  description?: string;
};
type UserThemeResponce = {
  id: number;
  user_id: number;
  theme: Theme;
  description?: string;
};
export { Theme, UserThemeResponce };
