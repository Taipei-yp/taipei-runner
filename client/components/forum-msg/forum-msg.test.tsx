import React from "react";
import TestRenderer from "react-test-renderer";
import { Msg } from "client/models/msg";
import { ForumMsg } from ".";

describe("Component ForumMsg", () => {
  const testMgs: Msg = {
    id: 1,
    author: {
      firstName: "test name",
      lastName: "test name",
      avatar: "/avarar",
    },
    createdAt: "12.12.2020",
    text: "test text",
  };

  const forumMsg = TestRenderer.create(<ForumMsg msg={testMgs} />);

  it("correct render", () => {
    expect(forumMsg.toJSON()).toMatchSnapshot();
  });
});
