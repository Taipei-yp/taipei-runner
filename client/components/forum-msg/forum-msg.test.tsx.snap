// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ForumMsg correct render 1`] = `
<div
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <img
        alt="test name test name avatar"
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
        src="test-file-stub"
      />
    </div>
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <span
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
      >
        test name test name
      </span>
      <span
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
      >
        12.12.2020, 00:00:00
      </span>
    </div>
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <span
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
      >
        test text
      </span>
    </div>
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <button
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
        type="button"
      >
        <span
          className={
            Object {
              "has": [Function],
              "is": [Function],
              "mix": [Function],
              "split": [Function],
              "state": [Function],
              "toString": [Function],
            }
          }
        >
          ♥
        </span>
        <span
          className={
            Object {
              "has": [Function],
              "is": [Function],
              "mix": [Function],
              "split": [Function],
              "state": [Function],
              "toString": [Function],
            }
          }
        >
          (undefined)
        </span>
      </button>
      <button
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
        type="button"
      >
        <span
          className={
            Object {
              "has": [Function],
              "is": [Function],
              "mix": [Function],
              "split": [Function],
              "state": [Function],
              "toString": [Function],
            }
          }
        >
          ↳
        </span>
      </button>
    </div>
  </div>
</div>
`;
